code/core/src/controller/MainController.java

```diff
@@ -55,22 +55,35 @@ public abstract class MainController extends ScreenAdapter implements IOnLevelLo
      * @param delta Time since last loop.
      */
     @Override
-    public final void render(float delta) {
+    public void render(float delta) {
         if (doFirstFrame) {
             firstFrame();
         }
+        batch.setProjectionMatrix(camera.combined);
+        if (runLoop()) {
```

Comment:
```suggestion
        if (!runLoop()) { return; }
```
bei den anderen Ifs ebenso, das spart die Einrückungstiefe

Reply:
Das `return;` wird zwar gespart, aber dann ist die Verschachtelungstiefe zu hoch... an der Semantik ändert sich ja nix...

also ich fände das einfacher zu lesen.